src/Services/DataService/Epics/FetchRecordEpic.test.ts

```diff
@@ -0,0 +1,761 @@
+                fullText: expectedResult.fullText,
+              },
+            });
+            resolve();
+          } catch (e) {
+            reject(e);
+          }
+        }, secondTimeOutPeriod);
+      });
+    });
+
+    it("should emit a single action if not called with another buffer", () => {
+      const id = random.number().toString();
+
+      const bufferObservable = createBufferObservable(context)(id);
+
+      let action;
+      bufferObservable
+        .take(1)
+        .subscribe(a => action = a);
+
+      return new Promise((resolve, reject) => {
+        setTimeout(() => {
+          try {
+            expect(action).to.deep.equal({
+              type: "PUSH_RECORD",
+              payload: {
+                id,
+                fullText: expectedResult.fullText,
+              },
+            });
+            resolve();
+          } catch (e) {
+            reject(e);
+          }
+        }, DEFAULT_COALESCE_BUFFER_TIME + 1);
+      });
+    });
+
+  });
+
+  describe("performBufferedRequest", () => {
+
+    it("should emit a FETCH_ALL action if multiple items are requested within the given period", () => {
+      const firstId = random.number().toString();
+      const secondId = random.number().toString();
+      const thirdId = random.number().toString();
+      const fourthId = random.number().toString();
+      const firstTimeOutPeriod = DEFAULT_COALESCE_BUFFER_TIME + 1;
+      const secondTimeOutPeriod = DEFAULT_COALESCE_BUFFER_TIME + 2;
+
+      const performRequest = performBufferedRequest(context);
+      
+      const sut = performRequest(firstId);
+
+      let action;
+      sut
+        .take(1)
+        .subscribe(a => action = a);
+
+      performRequest(secondId);
+      performRequest(thirdId);
+
+      setTimeout(() => {
+        performRequest(fourthId);
+      }, firstTimeOutPeriod);
+
+      const queryBuilder = new QueryBuilder(context.name, { ids: [firstId, secondId, thirdId] });
+      return new Promise((resolve, reject) => {
+        setTimeout(() => {
+          try {
+            expect(action).to.deep.equal({
+              type: "FETCH_ALL",
+              payload: queryBuilder,
+            });
+            resolve();
+          } catch (e) {
+            reject(e);
+          }
+        }, secondTimeOutPeriod);
+      });
+    });
+
+    it("should return an empty observable when it's called subsequent times with the same context name within the timeout period", () => {
+      const id = random.number().toString();
+      const secondId = random.number().toString();
+
+      const performRequest = performBufferedRequest(context);
+      performRequest(id);
+
+      const spies = {
+        next: spy(),
+        complete: spy(),
+      };
+
+      performRequest(secondId)
+        .subscribe(spies);
+
+      expect(spies.next.callCount).to.equal(0, "it should not call next method when subscribing");
+      expect(spies.complete.callCount).to.equal(1, "it should complete immediately");
```

Comment:
this doesn't seem to test that it "returns an empty observable" as the description says, is that expected?

Reply:
oh I think I understand, you are subscribing to the empty observable returned when calling performRequest with the secondId... something about this test doesnt make it easy to see that, maybe some comments would be helpful to clarify and/or renaming "it" in the expect statement to "empty observable"?

Reply:
I think we should definitely change the name to make it more clear.  It was hard to test this directly, and so we're doing so by proxy, so to speak.  My understanding of `empty$` is that it makes an observable that is resolved immediately, and so this was the most effective way to do so.

Reply:
Oops, our comments crossed.  Yeah, great idea!